### PR TITLE
chore: App router migration - general settings

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

- Tackles the following settings categories
	- Profile
	- Security
	- Billing
	- Developer

- Adds missing pages in Admin, Billing and Developer sections
- Removes references to the pages routes
- Refactors how we handle the settings headers. Instead of relying on the `MetaContext` we now generate the settings header server side
    - In the `page.tsx` we get the session and t function server side.
    - We then use the `<SettingsHeader>` component to set the page title and description
    	- The `<SettingsHeader>` is abstracted from the `<SettingsLayout>`	

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- For the new routes visit the normal pages routers and the`/future` routes

### CODE REVIEW

**Comment by hbjORbj** [comment]
Hey @joeauyeung ! I opened a [PR](https://github.com/calcom/cal.com/pull/16337), fixing type errors in this branch

**Comment by anikdhabal** [comment]
> Hey @joeauyeung ! I opened a [PR](https://github.com/calcom/cal.com/pull/16337), fixing type errors in this branch

We need to fix the type error in this branch before merging, no need to create a new for that. @joeauyeung will fix the type error

**Comment by hbjORbj** [comment]
@anikdhabal Hey! My PR is targeting this branch so I think it's not a problem! Also, I wanted to push directly to this branch but I've got no GH permissions yet. I think @joeauyeung already asked the team to give me a maintainer role, so it's all good.

**Comment by hbjORbj** [comment]
Oh, I see you already closed the PR. That's ok, I will cherry-pick my commits and push to this branch once I get the GH permissions.

**Comment by anikdhabal** [comment]
> Oh, I see you @anikdhabal already closed the PR. That's ok, I will cherry-pick my commits and push to this branch once I get the GH permissions.

Hey @hbjORbj, nice to meet you. Let's push the changes in this branch once you have permission

**Review by keithwillcode — DISMISSED** [review]

File: apps/web/app/future/settings/admin/organizations/[id]/edit/page.tsx
Comment (hbjORbj) [reply apps/web/app/future/settings/admin/organizations/[id]/edit/page.tsx]: Thanks, addressed

**Comment by hbjORbj** [comment]
Hey @joeauyeung , I see that you introduce quite changes to some submodules in `/packages`; e.g., `/packages/trpc`. Can you leave some explanatory comments?

File: packages/features/settings/appDir/SettingsHeader.tsx
Comment (joeauyeung) [inline packages/features/settings/appDir/SettingsHeader.tsx:1]: This is the big change in this folder. This is abstracted from the `SettingsLayout`. This way we can pass the page heading data directly to the server component rather that relying on `<Meta />` and `MetaContext`

File: packages/lib/server/repository/organization.ts
Comment (joeauyeung) [inline packages/lib/server/repository/organization.ts:1]: We're aiming to slowly transition a lot of our logic into a repository pattern. For this PR I moved the logic from our `trpc` endpoints into the repository so it could be still be used in `trpc` and some server components if we need them.

**Review by joeauyeung — COMMENTED** [review]
@hbjORbj some page components live in the `/packages` folder so that's why you're seeing a lot of changes there. I did comment on some changes I've added.

File: apps/web/app/future/settings/admin/organizations/[id]/edit/page.tsx
Comment (hbjORbj) [inline apps/web/app/future/settings/admin/organizations/[id]/edit/page.tsx:32]: I think we can replace `redirect("/404")` with `notFound()`.

Comment (hbjORbj) [inline apps/web/app/future/settings/admin/organizations/[id]/edit/page.tsx:32]: I think we can replace `redirect("/404")` with `notFound()`.

**Review by hbjORbj — CHANGES_REQUESTED** [review]

File: apps/web/app/future/settings/my-account/calendars/page.tsx
Comment (hbjORbj) [inline apps/web/app/future/settings/my-account/calendars/page.tsx:36]: I think this should be `CalendarListContainer` instead? `CalendarsView` has the same CTA; e.g.,
```
const CalendarsView = () => {
  const { t } = useLocale();

  return (
    <>
      <Meta
        title={t("calendars")}
        description={t("calendars_description")}
        CTA={<AddCalendarButton />}
        borderInShellHeader={false}
      />
      <div className="mt-8">
        <CalendarListContainer />
      </div>
    </>
  );
};
```

**Review by hbjORbj — CHANGES_REQUESTED** [review]
Hey, Joe, great work! I left a comment. Also,
1. `future/settings/my-account/calendars` has a compile error
2. `future/settings/admin/lockedSMS` has a double layout as in the screenshot. This is because the `getLayout` in `settings/layout.tsx` is applied to `settings/admin/lockedSMS` (which also has its own layout: admin layout). If you want to avoid this complication, you can go back to using WithLayout in each page.
3. Can you add feature flag env vars? 
Screenshot 2024-09-04 at 22 51 42

**Comment by hbjORbj** [comment]
Also, if it helps, here's a list you can use to go through:

`settings/admin/lockedSMS` 
`settings/admin/organizations/[id]/edit` 
`settings/admin/users/[id]/edit` 
`settings/billing` 
`settings/developer/api-keys` 
`settings/developer/webhooks` 
`settings/developer/webhooks/new` 
`settings/developer/webhooks/[id]` 
`settings/my-account/appearance` 
`settings/my-account/calendars` 
`settings/my-account/conferencing` 
`settings/my-account/general` 
`settings/my-account/out-of-office` 
`settings/my-account/profile` 
`settings/organizations/admin-api` 
`settings/security/impersonation` 
`settings/security/password` 
`settings/security/sso` 
`settings/security/two-factor-auth`

**Comment by joeauyeung** [comment]
@hbjORbj the admin settings have a different layout than the other settings pages. What are your thoughts on having two route groups for the different layouts?

https://github.com/calcom/cal.com/pull/16312/commits/dcb2b6073fc50453e0620ea366a51623036aef2a

**Comment by joeauyeung** [comment]
@hbjORbj I also refactored the pages to remove instances of `<Meta />` from the app router pages. 

Two changes I made were:
- If the page was in the `/web/modules` folder and was just the `<Meta />` component and the page component. We can remove the page from the `/modules` and call the `<Meta />` component directly in the page route and import the page component directly in both
- In some skeleton loaders it would use the `<Meta />` component. This was used to set the meta on page load. However, since we can do this server side in the app router, I removed it from the skeleton loaders. I didn't see any noticeable differences between the two types of routers.

**Comment by hbjORbj** [comment]
@joeauyeung Looks great to me! One issue was that `SettingsLayoutAppDir` was not synced with `SettingsLayout` in terms of styling, so I addressed it in this [commit](https://github.com/calcom/cal.com/pull/16312/commits/b44332a844d9f8ac3505b264d23589cd32d3c5de).

`settings/admin/lockedSMS` ✅
`settings/admin/organizations/[id]/edit` ✅
`settings/admin/users/[id]/edit` ✅
`settings/billing` ✅
`settings/developer/api-keys` ✅
`settings/developer/webhooks` ✅
`settings/developer/webhooks/new` ✅
`settings/developer/webhooks/[id]` ✅
`settings/my-account/appearance` ✅
`settings/my-account/calendars` ✅
`settings/my-account/conferencing` ✅
`settings/my-account/general` ✅
`settings/my-account/out-of-office` ✅
`settings/my-account/profile` ✅
`settings/organizations/admin-api` ✅
`settings/security/impersonation` ✅
`settings/security/password` ✅
`settings/security/sso` ✅
`settings/security/two-factor-auth` ✅

**Review by hbjORbj — APPROVED** [review]
@joeauyeung Looks great to me! One issue was that `SettingsLayoutAppDir` was not synced with `SettingsLayout` in terms of styling, so I addressed it in this [commit](https://github.com/calcom/cal.com/pull/16312/commits/b44332a844d9f8ac3505b264d23589cd32d3c5de).

`settings/admin/lockedSMS` ✅
`settings/admin/organizations/[id]/edit` ✅
`settings/admin/users/[id]/edit` ✅
`settings/billing` ✅
`settings/developer/api-keys` ✅
`settings/developer/webhooks` ✅
`settings/developer/webhooks/new` ✅
`settings/developer/webhooks/[id]` ✅
`settings/my-account/appearance` ✅
`settings/my-account/calendars` ✅
`settings/my-account/conferencing` ✅
`settings/my-account/general` ✅
`settings/my-account/out-of-office` ✅
`settings/my-account/profile` ✅
`settings/organizations/admin-api` ✅
`settings/security/impersonation` ✅
`settings/security/password` ✅
`settings/security/sso` ✅
`settings/security/two-factor-auth` ✅

**Comment by hbjORbj** [comment]
@joeauyeung One UI issue I found is that settings pages of app router have a bottom border that doesn't exist in pages router

- pages router
Screenshot 2024-09-08 at 21 01 22


- app router
Screenshot 2024-09-08 at 21 01 16

**Pull request merged**
